frontend/cs/r1cs/api.go

```diff
@@ -576,3 +576,19 @@ func (system *r1cs) negateLinExp(l compiled.LinearExpression) compiled.LinearExp
 func (system *r1cs) Compiler() frontend.Compiler {
 	return system
 }
+
+func (system *r1cs) Commit(v ...frontend.Variable) (frontend.Variable, error) {
+
```

Comment:
need to ensure we do it only once? since it uses a singleton (system.CommitmentInfo). But ... technically we could have (if needed) a list of []CommitmentInfo right?

Reply:
It's definitely ideal for different subcircuits to be able to get different commitments. Maybe something like
```go
func Commit(uid []byte, v ...frontend.Variable) (frontend.Variable, error)
```
that could be called multiple times.
I think it's worth trying it but would involve even more modifying-a-hint-after-it's-created type shenanigans. I'll create an issue and once I'm a bit more secure that my current hint hacks are okay, I'll take a stab at it.

But in the meantime I agree there should be an error if `Commit` is called more than once.